flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -72,8 +73,8 @@ public T reduce(T value1, T value2) throws Exception {
 		for (int position : fields) {
 			// Save position of compared key
 			// Get both values - both implement comparable
-			Comparable comparable1 = value1.getFieldNotNull(position);
-			Comparable comparable2 = value2.getFieldNotNull(position);
+			Comparable comparable1 = ((Tuple)value1).getFieldNotNull(position);
```

Comment:
I just updated the PR with the return type ReduceOperator. I was trying to work with scala Tuples and tried creating an example like SelectByFunctionsTest.  TupleTypeInfo and related things lik TupleSerializer and Tuplecomparator all accepts only java based tuples and not scala tuples. So am not pretty sure on how to make this SelectByMaxFunction to work with scala tuples.  Is there any other code that makes thing work with both scala.Tuple and java tuples?